framework/include/mesh/MooseMesh.h

```diff
@@ -1594,7 +1594,7 @@ MooseMesh::buildTypedMesh(unsigned int dim)
     // Set custom partitioner
     if (!_custom_partitioner.get())
       mooseError("Custom partitioner requested but not set!");
-    mesh->partitioner().reset(_custom_partitioner.release());
+    mesh->partitioner() = _custom_partitioner->clone();
```

Comment:
This feels like _custom_partitioner should be a shared pointer. We will be creating new clones for every "newly minted" mesh objects. So for example if we have two cartesian meshes, we stitch them, extrude them one direction, extrude them again another direction, then we have 5 custom partitioners (in addition to the one held by the moosemesh)